integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/externalformat/CsvBasedCarbonTableSuite.scala

```diff
@@ -241,4 +243,170 @@ class CsvBasedCarbonTableSuite extends QueryTest
+         | deptname String, projectcode int, projectjoindate String,projectenddate String,
+         | doj String, workgroupcategory int, workgroupcategoryname String,deptno int,
+         | attendance String, utilization String,salary String)
+         | STORED BY 'carbondata'
+         | TBLPROPERTIES(
+         | 'format'='csv',
+         | 'csv.header'='eMPno, empname,designation, doj, workgroupcategory,
+         | workgroupcategoryname, deptno, deptname, projectcode, projectjoindate, projectenddate,
+         |  attendance, utilization, SALARY'
+         | )
+       """.stripMargin
+    )
+    val exception = intercept[MalformedCarbonCommandException] {
+      sql(s"alter table $csvCarbonTable drop columns(empname,empno)")
+    }
+    assert(exception.getMessage.contains("Unsupported operation on external table"))
+  }
+
+  test("block change datatype on external format table") {
+    sql(
+      s"""
+         | CREATE TABLE $csvCarbonTable(empname String, empno smallint, designation string,
+         | deptname String, projectcode int, projectjoindate String,projectenddate String,
+         | doj String, workgroupcategory int, workgroupcategoryname String,deptno int,
+         | attendance String, utilization String,salary String)
+         | STORED BY 'carbondata'
+         | TBLPROPERTIES(
+         | 'format'='csv',
+         | 'csv.header'='eMPno, empname,designation, doj, workgroupcategory,
+         | workgroupcategoryname, deptno, deptname, projectcode, projectjoindate, projectenddate,
+         |  attendance, utilization, SALARY'
+         | )
+       """.stripMargin
+    )
+    val exception = intercept[MalformedCarbonCommandException] {
+      sql(s"alter table $csvCarbonTable change empno empno bigint")
+    }
+    assert(exception.getMessage.contains("Unsupported operation on external table"))
+  }
+
+  test("block create external format table if columns not in format.header") {
+    val exception = intercept[ProcessMetaDataException] {
+      sql(
+        s"""
+           | CREATE TABLE $csvCarbonTable(empname String, empno smallint, designation string,
+           | deptname String, projectcode int, projectjoindate String,projectenddate String,
+           | doj String, workgroupcategory int, workgroupcategoryname String,deptno int,
+           | attendance String, utilization String,salary String)
+           | STORED BY 'carbondata'
+           | TBLPROPERTIES(
+           | 'format'='csv',
+           | 'csv.header'='empname, designation, doj,
+           | workgroupcategory, workgroupcategoryname, deptno,
+           | deptname, projectcode, projectjoindate, projectenddate,
+           | attendance, utilization, SALARY',
+           | 'csv.delimiter'='|'
+           | )
+       """.stripMargin
+      )
+    }
+    assert(exception.getMessage.contains("column 'empno' not in tblproperty 'csv.header'"))
+  }
+
+  test("test desc formatted for external format table") {
+    sql(
```

Comment:
better to extract the createTableCommand to a function to reduce duplicated code

Reply:
fixed